botocore/httpchecksum.py

```diff
@@ -410,6 +435,19 @@ def _apply_request_trailer_checksum(request):
         headers["Content-Encoding"] = "aws-chunked"
     headers["X-Amz-Trailer"] = location_name
 
+    checksum_algorithm_name = algorithm["algorithm"]
```

Comment:
This looks identical to the logic above, I'd prefer we consolidate this to one location which makes it easier to maintain. For example, if we make a change in the future we only have to remember to make it in one place.